torch/nn/modules/linear.py

```diff
@@ -122,6 +123,27 @@ def reset_parameters(self) -> None:
             init.uniform_(self.bias, -bound, bound)
 
     def forward(self, input: Tensor) -> Tensor:
+        if not torch.is_grad_enabled() and not self.packed:
+            # Pack weights
+            if torch._C._has_mkldnn:
+                packed_weight = torch._C._nn.pack_linear(self.weight, input)
+                if packed_weight is not None:
+                    self.weight = torch.nn.Parameter(packed_weight)
+            self.packed = True
+
+        if torch.is_grad_enabled() and self.packed:
+            # Unpack weights
+            if torch._C._has_mkldnn:
+                if self.weight.is_mkldnn:
```

Comment:
And what end user is supposed to do, if they load serialized model with pre-packed weights, but want to unpack it on the system that does not have mkldnn enabled?